src/components/web-monitoring-ui.jsx

```diff
@@ -48,70 +62,125 @@ export default class WebMonitoringUi extends React.Component {
       });
   }
 
   componentWillMount () {
-    this.loadPages();
+    this.loadUser();
   }
 
   render () {
-    const withData = bindComponent({pages: this.state.pages, user: this.state.user});
-    const modal = this.state.showLogin ? this.renderLoginDialog() : null;
+    const {
+      showLogin,
+      user,
+      isLoading,
+      currentFilter
+    } = this.state;
+
+    if (isLoading) {
+      return <Loading />
+    }
+
+    if (!user) {
+      return this.renderLoginDialog("You must be logged in to view pages");
+    }
+
+    const withData = (ComponentType, pageType) => {
+      return (routeProps) => {
+        const pages = this.state[pageType];
+        if (!pages) {
+          this.loadPages(pageType === 'pages');
+        }
+        return <ComponentType
+            {...routeProps}
+            pages={pages}
+            user={this.state.user}
+          />;
+      };
+    };
+    const modal = showLogin ? this.renderLoginDialog() : null;
 
     return (
       <div>
         <Router>
           <div id="application">
-            <NavBar title="EDGI" user={this.state.user} showLogin={this.showLogin} logOut={this.logOut} />
-            <Route exact path="/" render={withData(PageList)} />
-            <Route path="/page/:pageId/:change?" render={withData(PageDetails)} />
+            <NavBar title="EDGI"
+              user={this.state.user}
+              showLogin={this.showLogin}
+              logOut={this.logOut}
+              setCurrentFilter={this.setCurrentFilter}
+              currentFilter={this.state.currentFilter}
+            />
+            <div>
+              <Route exact path="/" render={() => (
+                this.state.user
+                  ? (<Redirect to="/assignedPages" />)
+                  : (<Redirect to="/pages" />)
+              )}/>
+              <Route path="/pages" render={withData(PageList, 'pages')} />
+              <Route path="/assignedPages" render={withData(PageList, 'assignedPages')} />
+              <Route path="/page/:pageId/:change?" render={
+                  (routeProps) => <PageDetails
+                    {...routeProps}
+                    user={user}
+                    currentFilter={currentFilter}
+                    pages={this.state[currentFilter]} />
+                } />
+            </div>
           </div>
         </Router>
         {modal}
       </div>
     );
   }
 
-  renderLoginDialog () {
+  renderLoginDialog (message) {
     return (
       <AriaModal
-        titleText="Log in"
+        titleText="Log In"
         onExit={this.hideLogin}
         applicationNode={document.getElementById('web-monitoring-ui-root')}
         dialogClass="dialog__body"
         underlayClass="dialog dialog__underlay"
         verticallyCenter={true}
       >
-        <LoginForm cancelLogin={this.hideLogin} onLogin={this.afterLogin} />
+        <LoginForm message={message} cancelLogin={this.hideLogin} onLogin={this.afterLogin} />
```

Comment:
Message doesn't have to be prop or part of api, yet or ever? Just code into Login component